Sally7/S7Connection.cs

```diff
@@ -49,6 +49,11 @@ public void Close()
             client.Close();
         }
 
+        public void Dispose()
+        {
+            Close();
```

Comment:
@mycroes Since we are targeting .NET 4.5.2 it's not possible, as it was [protected prior to .NET 4.6](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.tcpclient.dispose?f1url=https%3A%2F%2Fmsdn.microsoft.com%2Fquery%2Fdev15.query%3FappId%3DDev15IDEF1%26l%3DEN-US%26k%3Dk(System.Net.Sockets.TcpClient.Dispose)%3Bk(DevLang-csharp)%26rd%3Dtrue%26f%3D255%26MSPPError%3D-2147217396&view=netframework-4.5.2)

I avoid #ifdef when I can, and since the close method of the client does the dispose, I mimicked it's way to close / dispose, instead of duplicating the close method content, in case things are added in the close method.

Reply:
Good point, I actually care about net452. However, `TcpClient` does implement `IDisposable` in net452 so we can do this:
```suggestion
            ((IDisposable) client).Dispose();
```

Reply:
@mycroes good point! I'll fix this in a new PR with a better message.